specs/lightclient/beacon-chain.md

```diff
@@ -151,13 +172,13 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
 ```python
 def process_sync_committee(state: BeaconState, body: BeaconBlockBody) -> None:
     # Verify sync committee aggregate signature signing over the previous slot block root
-    previous_slot = max(state.slot, Slot(1)) - Slot(1)
+    previous_slot = max(Slot(state.slot), Slot(1)) - Slot(1)
     committee_indices = get_sync_committee_indices(state, get_current_epoch(state))
     participant_indices = [committee_indices[i] for i in range(len(committee_indices)) if body.sync_committee_bits[i]]
     participant_pubkeys = [state.validators[participant_index].pubkey for participant_index in participant_indices]
     domain = get_domain(state, DOMAIN_SYNC_COMMITTEE, compute_epoch_at_slot(previous_slot))
     signing_root = compute_signing_root(get_block_root_at_slot(state, previous_slot), domain)
-    assert bls.FastAggregateVerify(participant_pubkeys, signing_root, body.sync_committee_signature)
+    assert optional_fast_aggregate_verify(participant_pubkeys, signing_root, body.sync_committee_signature)
```

Comment:
Using the word "optional" in a verification function in a spec that people are going to read is a bit misleading imo....

Is the goal of the `optional_fast_aggregate_verify` to allow the verification to be bypassed if we want to run tests without the BLS overhead? If so it may be worth not calling the function optional despite its role; intuitiveness to a reader trying to understand is more important than intuitiveness to someone directly working on the executable spec tooling.

Reply:
`optional_fast_aggregate_verify` is to allow `len(participant_pubkeys) == 0` case since it's illegal in IETF BLS draft standard APIs right now. The word "optional" may be too mypy-ish and not readable enough. 😅  Any naming suggestion?

@JustinDrake mentioned that it's possible that BLST may implement this feature at the API level. Then maybe we can import/refer to another document from the BLS standard region.

Reply:
Why not just call it `fast_aggregate_verify`?

It's the BLS standard that's implementing a weird quirky method with a special case, not us; we're just doing aggregate verification with the mathematically obvious generalization to the empty pubkey list case. So IMO it's totally okay for us to just call it fast aggregate verification.

Reply:
Agreed that `optional_` is awkward. One possibility is to call it `eth2_fast_aggregate_verify`. On the one hand it avoids confusion with the IETF `bls.FastAggregateVerify` and on the other hand the naming suggests it is an Eth2 variant.

Reply:
Thank you both!

I'm inclined to `eth2_fast_aggregate_verify` to avoid confusion. 👍